test/Streamly/Test/FileSystem/Event.hs

```diff
@@ -8,54 +8,63 @@
 -- Portability : GHC
 --
 -- Just report all events under the paths provided as arguments
-
 module Main (main) where
 
-import Control.Monad.IO.Class (MonadIO)
 import Data.Function ((&))
+import System.Environment (getArgs)
+import Streamly.Prelude (SerialT)
 import Data.List.NonEmpty (NonEmpty)
+#if !defined(CABAL_OS_WINDOWS)   
+import Control.Monad.IO.Class (MonadIO)    
+import qualified Streamly.Unicode.Stream as Unicode
+import qualified Streamly.Internal.Data.Array.Storable.Foreign as Array
 import Data.Word (Word8)
 import Streamly.Internal.Data.Array.Storable.Foreign (Array)
-import System.Environment (getArgs)
-import Streamly.Prelude (SerialT)
+#endif
 
 import qualified Data.List.NonEmpty as NonEmpty
-import qualified Streamly.Unicode.Stream as Unicode
 import qualified Streamly.Internal.Data.Stream.IsStream as Stream
 #if defined(CABAL_OS_DARWIN)
 import qualified Streamly.Internal.FileSystem.Event.Darwin as Event
 #elif defined(CABAL_OS_LINUX)
 import qualified Streamly.Internal.FileSystem.Event.Linux as Event
-#else
-#error "FS Events not supported on this platform
+#elif defined(CABAL_OS_WINDOWS)
+import qualified Streamly.Internal.FileSystem.Event.Windows as Event
 #endif
-import qualified Streamly.Internal.Data.Array.Storable.Foreign as Array
 
 -------------------------------------------------------------------------------
 -- Utilities
 -------------------------------------------------------------------------------
-
+#if !defined(CABAL_OS_WINDOWS)
 toUtf8 :: MonadIO m => String -> m (Array Word8)
 toUtf8 = Array.fromStream . Unicode.encodeUtf8 . Stream.fromList
-
+#endif
 -------------------------------------------------------------------------------
 -- Main
 -------------------------------------------------------------------------------
-
-watchPaths :: NonEmpty (Array Word8) -> SerialT IO Event.Event
-
-#if defined(CABAL_OS_DARWIN)
+#if defined(CABAL_OS_WINDOWS)
+watchPaths :: NonEmpty FilePath -> SerialT IO Event.Event
+watchPaths = Event.watchTrees
+#elif defined(CABAL_OS_DARWIN)
+watchPaths :: NonEmpty (Array Word8) -> SerialT IO Event.Event    
 watchPaths = Event.watchTrees
 #elif defined(CABAL_OS_LINUX)
+watchPaths :: NonEmpty (Array Word8) -> SerialT IO Event.Event    
 watchPaths = Event.watchPaths
-#else
-#error "Unsupported platform
 #endif
 
+#if defined(CABAL_OS_WINDOWS)
+main :: IO ()
+main = do
+    args <- getArgs   
+    watchPaths (NonEmpty.fromList args) 
+        & Stream.mapM_ (putStrLn . Event.showEvent)
+
+#else    
```

Comment:
Instead of using the whole code under `CABAL_OS_WINDOWS` we can just put the conditional around the `paths <- mapM toUtf8 args` in the code below.